src/include/cart/api.h

```diff
@@ -694,162 +692,142 @@ crt_ep_abort(crt_endpoint_t *ep);
 #define CRT_GEN_GET_NAME(seq) BOOST_PP_SEQ_ELEM(1, seq)
 #define CRT_GEN_GET_KIND(seq) BOOST_PP_SEQ_TAIL(BOOST_PP_SEQ_TAIL(seq))
 
-#define CRT_GEN_STRUCT_FIELD(r, data, seq)				\
-	BOOST_PP_IF(BOOST_PP_EQUAL(CRT_ARRAY, CRT_GEN_GET_KIND(seq)),	\
-		struct {						\
-			uint64_t		 ca_count;		\
-			CRT_GEN_GET_TYPE(seq)	*ca_arrays;		\
-		},							\
-		CRT_GEN_GET_TYPE(seq))					\
-	BOOST_PP_IF(BOOST_PP_EQUAL(CRT_PTR, CRT_GEN_GET_KIND(seq)),	\
-		*CRT_GEN_GET_NAME(seq), CRT_GEN_GET_NAME(seq));
-
-#define CRT_GEN_STRUCT(struct_type_name, seq)				\
-	struct struct_type_name {					\
-		BOOST_PP_SEQ_FOR_EACH(CRT_GEN_STRUCT_FIELD, , seq)	\
+#define CRT_GEN_STRUCT_FIELD(r, data, seq)                                                         \
```

Comment:
(style)  Macros with multiple statements should be enclosed in a do - while loop
(style)  macros should not use a trailing semicolon